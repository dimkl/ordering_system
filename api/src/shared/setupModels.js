const { Model } = require('objection');
const knex = require('./knex');

module.exports = () => { Model.knex(knex) }