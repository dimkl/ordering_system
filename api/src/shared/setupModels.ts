import { Model } from "objection";
import { knex } from "./knex";

export default () => {
  Model.knex(knex);
  return knex;
};
