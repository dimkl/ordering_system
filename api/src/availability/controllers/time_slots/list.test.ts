/**
 * @integration-test true
 * @data-factory true
 */
import type { Knex } from "knex";

import setupModels from "../../../shared/setupModels";

describe("GET /time_slots", () => {
  let knex: Knex;
  beforeAll(() => (knex = setupModels()));
  beforeEach(() =>
    knex.raw("truncate orders, order_items, customers, users, products, time_slots, slots cascade;")
  );
  afterAll(() => knex.destroy());

  it("returns all time_slots", async () => {
    await DataFactory.createTimeSlot();

    const response = await request.get("/time_slots").set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body.length).toBe(1);
    expect(response.body[0]).toMatchSnapshot({
      id: expect.any(Number),
      slot_id: expect.any(String),
      customer_id: expect.any(String),
      uuid: expect.any(String),
      started_at: expect.any(String),
      created_at: expect.any(String),
      updated_at: expect.any(String)
    });
  });

  it("returns specified time_slot", async () => {
    const timeSlot = await DataFactory.createTimeSlot();

    const response = await request
      .get("/time_slots/" + timeSlot.id)
      .set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({
      id: expect.any(Number),
      slot_id: expect.any(String),
      customer_id: expect.any(String),
      uuid: expect.any(String),
      started_at: expect.any(String),
      created_at: expect.any(String),
      updated_at: expect.any(String)
    });
  });

  it("returns specified time_slot using uuid", async () => {
    const timeSlot = await DataFactory.createTimeSlot();

    const response = await request
      .get("/time_slots/" + timeSlot.uuid)
      .set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body).toMatchSnapshot({
      id: expect.any(Number),
      slot_id: expect.any(String),
      customer_id: expect.any(String),
      uuid: expect.any(String),
      started_at: expect.any(String),
      created_at: expect.any(String),
      updated_at: expect.any(String)
    });
  });

  it("returns empty list of time_slots when there are no time_slots", async () => {
    const response = await request.get("/time_slots").set("Accept", "application/json");

    expect(response.status).toBe(200);
    expect(response.body).toEqual([]);
  });

  it("throws 404 when specified time_slot does not exist", async () => {
    const response = await request
      .get("/time_slots/" + Math.floor(Math.random() * 100))
      .set("Accept", "application/json");

    expect(response.status).toBe(404);
  });
});
