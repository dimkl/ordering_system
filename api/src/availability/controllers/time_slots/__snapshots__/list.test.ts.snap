// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /time_slots returns all time_slots 1`] = `
{
  "created_at": Any<String>,
  "customer_id": Any<String>,
  "ended_at": null,
  "id": Any<Number>,
  "slot_id": Any<String>,
  "started_at": Any<String>,
  "updated_at": Any<String>,
  "uuid": Any<String>,
}
`;

exports[`GET /time_slots returns specified time_slot 1`] = `
{
  "created_at": Any<String>,
  "customer_id": Any<String>,
  "ended_at": null,
  "id": Any<Number>,
  "slot_id": Any<String>,
  "started_at": Any<String>,
  "updated_at": Any<String>,
  "uuid": Any<String>,
}
`;

exports[`GET /time_slots returns specified time_slot using uuid 1`] = `
{
  "created_at": Any<String>,
  "customer_id": Any<String>,
  "ended_at": null,
  "id": Any<Number>,
  "slot_id": Any<String>,
  "started_at": Any<String>,
  "updated_at": Any<String>,
  "uuid": Any<String>,
}
`;
